tests/visualization_tests/test_slice.py

```diff
@@ -1,99 +1,364 @@
+        [prepare_study_with_trials, ["param_a", "param_b", "param_c"]],
+        [prepare_study_with_trials, ["param_a", "param_b", "param_c", "param_d"]],
+        [prepare_study_with_trials, None],
+        [_create_study_with_log_scale_and_str_category_2d, None],
+        [_create_study_mixture_category_types, None],
+    ],
+)
+def test_plot_slice(
+    plot_slice: Callable[..., Any],
+    specific_create_study: Callable[[], Study],
+    params: Optional[List[str]],
+) -> None:
+
+    study = specific_create_study()
+    figure = plot_slice(study, params=params)
+    if isinstance(figure, go.Figure):
+        figure.write_image(BytesIO())
+    else:
+        plt.savefig(BytesIO())
+
+
+def test_target_is_none_and_study_is_multi_obj() -> None:
+
+    study = create_study(directions=["minimize", "minimize"])
+    with pytest.raises(ValueError):
+        _get_slice_plot_info(study, None, None, "Objective Value")
+
+
+@pytest.mark.parametrize(
+    "specific_create_study",
+    [create_study, _create_study_with_failed_trial],
+)
+@pytest.mark.parametrize(
+    "params",
+    [
+        [],
+        ["param_a"],
+        ["param_a", "param_b"],
+        ["param_a", "param_b", "param_c"],
+        ["param_a", "param_b", "param_c", "param_d"],
+        None,
+    ],
+)
+def test_get_slice_plot_info_empty(
+    specific_create_study: Callable[[], Study], params: Optional[List[str]]
+) -> None:
 
-    # Plot a parameter.
-    figure = plot_slice(study, params=["y_log"])
-    assert figure.layout["xaxis_type"] == "log"
-    figure = plot_slice(study, params=["x_linear"])
-    assert figure.layout["xaxis_type"] is None
+    study = specific_create_study()
+    info = _get_slice_plot_info(study, params=params, target=None, target_name="Objective Value")
+    assert len(info.subplots) == 0
 
-    # Plot multiple parameters.
-    figure = plot_slice(study)
-    assert figure.layout["xaxis_type"] is None
-    assert figure.layout["xaxis2_type"] == "log"
+
+def test_get_slice_plot_info_non_exist_param_error() -> None:
+    study = prepare_study_with_trials()
+
+    with pytest.raises(ValueError):
+        _get_slice_plot_info(study, params=["optuna"], target=None, target_name="Objective Value")
+
+
+@pytest.mark.parametrize(
+    "params",
+    [
+        ["param_a"],
+        ["param_a", "param_b"],
+        ["param_a", "param_b", "param_c"],
+        ["param_a", "param_b", "param_c", "param_d"],
+        None,
+    ],
+)
+def test_get_slice_plot_info_params(params: Optional[List[str]]) -> None:
+
+    study = prepare_study_with_trials()
+    n_params = len(params) if params is not None else 4
+
+    info = _get_slice_plot_info(study, params=params, target=None, target_name="Objective Value")
+    assert len(info.subplots) == n_params
+
+
+@pytest.mark.parametrize(
+    "params",
+    [
+        ["param_a", "param_b"],
+        ["param_a", "param_b", "param_c"],
+    ],
+)
+def test_get_slice_plot_info_customized_target(params: List[str]) -> None:
+
```

Comment:
```suggestion
def test_get_slice_plot_info_customized_target() -> None:
    params = ["param_a", "param_b"]
```
We don't need this parametrization.